### ISSUE TITLE
Downloading cover art fails if too many files selected

### ISSUE THREAD

**Issue by MisterTea** [OpenEmu/OpenEmu#1658]
If I choose a handful of files and try to download cover art, it downloads with no problem.  If I select my entire library (1000s of files) and try to download all cover art, none of the art downloads.

**Comment by clobber** [comment]
If it's just NES, be sure it's just the 'good' versions of the ROM dumps, as bad dumps, over dumps, hacks, translations, etc won't pull cover art.

**Comment by MisterTea** [comment]
@clobber Hey, I don't think you understand this issue.  if I download the cover art for a few files, it works perfectly.  When I select a lot of files (e.g. all my nes roms), it starts out fine, but then at some point it starts failing and it will continue failing until it's done.  If I go back and re-select some of the failed roms, the cover art downloads fine.  It just can't handle a large selection.

**Comment by clobber** [comment]
I didn't understand because you went back and said it "seems to be isolated to NES games", which made me think it was just files not tracked in the database. There can't be a per-system issue when it comes to downloading covers unless it's a system that doesn't actually contain covers in the database.

If this is a legitimate issue with queuing up downloads of 1000's of covers I'll reopen, but not something we'll really look into soon.

**Comment by cyco** [comment]
@MisterTea could you check the artworks folder too see if new items show up? Should be located at ~/Library/Application Support/OpenEmu/Game Library/Artwork/.

Big fan of programming throwdown btw ;)

**Comment by MisterTea** [comment]
@cyco glad you like the show :-).  I'll test this out over the weekend and let you know, but my guess is that the new items will not be in ~/Library/Application Support/OpenEmu/Game Library/Artwork/ because they don't show up if I restart openemu.

It's almost as if the way openemu fetches the album art is not freeing some resources fast enough and after a few thousand fetches it starts giving up.

**Comment by toholio** [comment]
I think I have a related problem. Attempting to download a single SNES game's art fails, or simply doesn't do anything. If you again attempt to download a SNES game's art the contextual menu gets stuck on the screen. At that point you have to kill OpenEmu.

After reopening OpenEmu that same problem is still exists.

Attempting to delete a SNES game from the library also causes the contextual menu to get stuck on screen.

**Comment by clobber** [comment]
@toholio No, not related. Anyway, your version of OS X? Your version of OpenEmu?

Context menu sounds like 10.7/10.8 issue or something we've already patched in source, available in next release. In order for SNES ROMs to be identified, they need to not contain the [useless 512-byte copier header](http://romhack.wikia.com/wiki/SMC_header) (usually found on dumps with the .SMC file extension, you'll instead want ones with .SFC).

**Comment by toholio** [comment]
@clobber MacOS 10.10.4 and OpemEmu 1.0.4. Thanks for that. These are ROMs with a SMC extension so that's probably it.
